File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/dagger/AppComponent.java
package com.dbottillo.mtgsearchfree.dagger;

import com.dbottillo.mtgsearchfree.MTGApp;
import com.dbottillo.mtgsearchfree.model.database.SetDataSource;
import com.dbottillo.mtgsearchfree.model.storage.CardsHelper;
import com.dbottillo.mtgsearchfree.model.storage.CardsPreferences;
import com.dbottillo.mtgsearchfree.model.storage.CardsStorage;
import com.dbottillo.mtgsearchfree.model.storage.DecksStorage;
import com.dbottillo.mtgsearchfree.model.storage.GeneralData;
import com.dbottillo.mtgsearchfree.model.storage.PlayersStorage;
import com.dbottillo.mtgsearchfree.model.storage.ReleaseNoteStorage;
import com.dbottillo.mtgsearchfree.model.storage.SavedCardsStorage;
import com.dbottillo.mtgsearchfree.ui.BasicActivity;
import com.dbottillo.mtgsearchfree.ui.lifecounter.LifeCounterPresenterImpl;
import com.dbottillo.mtgsearchfree.util.DialogUtil;
import com.dbottillo.mtgsearchfree.util.FileManager;
import com.dbottillo.mtgsearchfree.util.FileUtil;
import com.dbottillo.mtgsearchfree.util.Logger;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AndroidModule.class, DataModule.class})
public interface AppComponent {

    CardsStorage getCardsStorage();

    SavedCardsStorage getSavedCardsStorage();

    SetDataSource getSetDataSource();

    PlayersStorage getPlayerStorage();

    DecksStorage getDecksStorage();

    CardsPreferences getCardsPreferences();

    GeneralData getGeneralPreferences();

    FileUtil getFileUtil();

    FileManager getFileManager();

    Logger getLogger();

    DialogUtil getDialogUtil();

    CardsHelper getCardsHelper();

    ReleaseNoteStorage getReleaseNoteStorage();

    void inject(MTGApp app);

    void inject(BasicActivity mainActivity);

    void inject(LifeCounterPresenterImpl playerPresenter);
}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/component/AppComponent.kt
package com.dbottillo.mtgsearchfree.component

import android.content.SharedPreferences
import com.dbottillo.mtgsearchfree.base.MTGApp
import com.dbottillo.mtgsearchfree.interactors.CardFilterInteractor
import com.dbottillo.mtgsearchfree.interactors.CardsInteractor
import com.dbottillo.mtgsearchfree.interactors.SetsInteractor
import com.dbottillo.mtgsearchfree.model.storage.CardFilterStorage
import com.dbottillo.mtgsearchfree.model.storage.CardsStorage
import com.dbottillo.mtgsearchfree.model.storage.SetsStorage
import com.dbottillo.mtgsearchfree.modules.AndroidModule
import com.dbottillo.mtgsearchfree.modules.DataModule
import com.dbottillo.mtgsearchfree.modules.InteractorsModule
import com.dbottillo.mtgsearchfree.view.activities.BasicActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidModule::class, DataModule::class, InteractorsModule::class))
interface AppComponent {

    fun getCardFilterInteractor(): CardFilterInteractor

    fun getCardsInteractor(): CardsInteractor

    fun getCardFilterStorage(): CardFilterStorage

    fun getCardsStorage(): CardsStorage

    fun getSetsInteractor(): SetsInteractor

    fun getSetsStorage(): SetsStorage

    fun getSharedPreferences(): SharedPreferences

    fun inject(application: MTGApp)

    fun inject(mainActivity: BasicActivity)

}